src/libcollections/btree/map.rs

```diff
@@ -328,6 +332,50 @@ impl<K: Ord, V> BTreeMap<K, V> {
     /// assert_eq!(map.insert(37, "c"), Some("b"));
     /// assert_eq!(map[&37], "c");
     /// ```
+    ///
+    /// If we have a more complex key, calls to `insert()` will
+    /// not update the value of the key. For example:
+    ///
+    /// ```
+    /// use std::cmp::Ordering;
+    /// use std::collections::BTreeMap;
+    /// use std::hash::{Hash, Hasher};
+    /// 
+    /// #[derive(Debug)]
+    /// struct Foo {
+    ///     a: u32,
+    ///     b: &'static str,
+    /// }
+    /// 
+    /// // we will compare `Foo`s by their `a` value only.
+    /// impl PartialEq for Foo {
+    ///     fn eq(&self, other: &Self) -> bool { self.a == other.a }
+    /// }
+    /// 
+    /// impl Eq for Foo {}
+    /// 
+    /// // we will hash `Foo`s by their `a` value only.
+    /// impl Hash for Foo {
+    ///     fn hash<H: Hasher>(&self, h: &mut H) { self.a.hash(h); }
+    /// }
+    /// 
+    /// impl PartialOrd for Foo {
+    ///     fn partial_cmp(&self, other: &Self) -> Option<Ordering> { self.a.partial_cmp(&other.a) }
+    /// }
+    /// 
+    /// impl Ord for Foo {
+    ///     fn cmp(&self, other: &Self) -> Ordering { self.a.cmp(&other.a) }
+    /// }
+    /// 
+    /// let mut map = BTreeMap::new();
+    /// map.insert(Foo { a: 1, b: "baz" }, ());
```

Comment:
This example may be a bit clearer (at least, help resolve the ambiguity above) if the value is non-trivial and checked, to see the change.